kafka-client/src/commonMain/kotlin/com/icemachined/kafka/common/serialization/JsonDeserializer.kt

```diff
@@ -0,0 +1,29 @@
+package com.icemachined.kafka.common.serialization
+
+import com.icemachined.kafka.clients.consumer.Headers
+import com.icemachined.kafka.common.header.KafkaHeaders
+
+import kotlin.reflect.KType
+import kotlinx.cinterop.toKString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+
+/**
+ * JsonDeserializer
+ */
+class JsonDeserializer<T>(
+    private val typesMap: Map<String, KType>
```

Comment:
suggestion to add factory methods for a single `KType` and with vararg `KType`

Reply:
what do you mean by vararg ktype factory?

Reply:
I extract resolvers

Reply:
> What do you mean by vararg ktype factory?
I mean that typesMaps is Map which can be built by values only.
```
val typesMap = listOf(typeOf<DiktatSuite>(), typeOf<DetectSuite>()).associateBy { it.toString() }
```

Reply:
it.toString()  this dosn't always work sometimes i thins type codes can be obtained from different serializer, for example spring, which will contain java classnames, so user will have to set it manually to integrate with spring kafka services.

Reply:
ok fixed